### PR TITLE
INTMDB-583: Update cdk.sh to create L2 and L3 CDK project as well

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Updated the `cdk.sh` to generate L1, L2 and L3 CDK contractors. 

```bash
# This shell script can be use to generate one L1 | L2 | L3 CDK Construct.
# In case of L1 constructor, the script uses the CFN template to generate the CDK resource.
#
# How to use it:
#   1. CDK L1:  ./cdk.sh "<RESOURCE NAME>" L1
#   1. CDK L1:  ./cdk.sh "<RESOURCE NAME>" L1
#   2. CDK L2:  ./cdk/sh "<RESOURCE NAME>" L2
#   4. CDK L3:  ./cdk/sh "<RESOURCE NAME>" L3
```

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

<details>
  <summary>Create L1 CDK constructor</summary>
  
 
```bash
./cdk.sh cluster L1                                                                                                                16.18.0 bazel 5.4.0
Generating L1 CDK resource
generating for MongoDB::Atlas::Cluster
cluster
Unresolvable read-only property (attribute) MongoDB::Atlas::Cluster.ConnectionStrings/Standard
Unresolvable read-only property (attribute) MongoDB::Atlas::Cluster.ConnectionStrings/StandardSrv
Unresolvable read-only property (attribute) MongoDB::Atlas::Cluster.ConnectionStrings/Private
Unresolvable read-only property (attribute) MongoDB::Atlas::Cluster.ConnectionStrings/PrivateSrv
~/workspace/mongodbatlas-cloudformation-resources/cdk/cdk-resources/cluster ~/workspace/mongodbatlas-cloudformation-resources/cdk
👾 Project definition file was created at /Users/andrea.angiolillo/workspace/mongodbatlas-cloudformation-resources/cdk/cdk-resources/cluster/.projenrc.js
yarn install v1.22.19
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 25.24s.
yarn install v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 2.81s.

> @mongodbatlas-awscdk/cluster@0.0.0 eslint
> npx projen eslint

Initialized empty Git repository in /Users/andrea.angiolillo/workspace/mongodbatlas-cloudformation-resources/cdk/cdk-resources/cluster/.git/
[main (root-commit) d47118a] chore: project created with projen
 20 files changed, 8977 insertions(+)
 create mode 100644 .eslintrc.json
 create mode 100644 .gitattributes
 create mode 100644 .github/pull_request_template.md
 create mode 100644 .github/workflows/build.yml
 create mode 100644 .github/workflows/pull-request-lint.yml
 create mode 100644 .github/workflows/release.yml
 create mode 100644 .github/workflows/upgrade-master.yml
 create mode 100644 .gitignore
 create mode 100644 .mergify.yml
 create mode 100644 .npmignore
 create mode 100644 .projen/deps.json
 create mode 100644 .projen/files.json
 create mode 100644 .projen/tasks.json
 create mode 100644 .projenrc.js
 create mode 100644 LICENSE
 create mode 100644 README.md
 create mode 100644 package.json
 create mode 100644 src/index.ts
 create mode 100644 tsconfig.dev.json
 create mode 100644 yarn.lock

```

</details>

<details>
  <summary>Create L2 CDK constructor</summary>
  
 
```bash
./cdk.sh testCDK L2                                                                                                                    16.18.0 bazel 5.4.0
Generating L2 CDK resource
~/workspace/mongodbatlas-cloudformation-resources/cdk/l2-cdk-resources/testCDK ~/workspace/mongodbatlas-cloudformation-resources/cdk
👾 Project definition file was created at /Users/andrea.angiolillo/workspace/mongodbatlas-cloudformation-resources/cdk/l2-cdk-resources/testCDK/.projenrc.js
yarn install v1.22.19
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

success Saved lockfile.
✨  Done in 18.57s.
yarn install v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

success Saved lockfile.
✨  Done in 2.81s.

> @mongodbatlas-awscdk-l2/testCDK@0.0.0 eslint
> npx projen eslint

Initialized empty Git repository in /Users/andrea.angiolillo/workspace/mongodbatlas-cloudformation-resources/cdk/l2-cdk-resources/testCDK/.git/
[main (root-commit) af74c6c] chore: project created with projen
 19 files changed, 8072 insertions(+)
 create mode 100644 .eslintrc.json
 create mode 100644 .gitattributes
 create mode 100644 .github/pull_request_template.md
 create mode 100644 .github/workflows/build.yml
 create mode 100644 .github/workflows/pull-request-lint.yml
 create mode 100644 .github/workflows/release.yml
 create mode 100644 .github/workflows/upgrade-master.yml
 create mode 100644 .gitignore
 create mode 100644 .mergify.yml
 create mode 100644 .npmignore
 create mode 100644 .projen/deps.json
 create mode 100644 .projen/files.json
 create mode 100644 .projen/tasks.json
 create mode 100644 .projenrc.js
 create mode 100644 LICENSE
 create mode 100644 README.md
 create mode 100644 package.json
 create mode 100644 tsconfig.dev.json
 create mode 100644 yarn.lock
```

</details>

<details>
  <summary>Create L3 CDK constructor</summary>
  
 
```bash
./cdk.sh testCDK L3                                                                                                            26s  16.18.0 bazel 5.4.0
Generating L3 CDK resource
~/workspace/mongodbatlas-cloudformation-resources/cdk/l3-cdk-resources/testCDK ~/workspace/mongodbatlas-cloudformation-resources/cdk
👾 Project definition file was created at /Users/andrea.angiolillo/workspace/mongodbatlas-cloudformation-resources/cdk/l3-cdk-resources/testCDK/.projenrc.js
yarn install v1.22.19
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

success Saved lockfile.
✨  Done in 16.43s.
yarn install v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

success Saved lockfile.
✨  Done in 3.07s.

> @mongodbatlas-awscdk-l3/testCDK@0.0.0 eslint
> npx projen eslint

Initialized empty Git repository in /Users/andrea.angiolillo/workspace/mongodbatlas-cloudformation-resources/cdk/l3-cdk-resources/testCDK/.git/
[main (root-commit) b0b0cb3] chore: project created with projen
 19 files changed, 8072 insertions(+)
 create mode 100644 .eslintrc.json
 create mode 100644 .gitattributes
 create mode 100644 .github/pull_request_template.md
 create mode 100644 .github/workflows/build.yml
 create mode 100644 .github/workflows/pull-request-lint.yml
 create mode 100644 .github/workflows/release.yml
 create mode 100644 .github/workflows/upgrade-master.yml
 create mode 100644 .gitignore
 create mode 100644 .mergify.yml
 create mode 100644 .npmignore
 create mode 100644 .projen/deps.json
 create mode 100644 .projen/files.json
 create mode 100644 .projen/tasks.json
 create mode 100644 .projenrc.js
 create mode 100644 LICENSE
 create mode 100644 README.md
 create mode 100644 package.json
 create mode 100644 tsconfig.dev.json
 create mode 100644 yarn.lock
```

</details>

<details>
  <summary>Wrong usage of the script</summary>

```bash

 ./cdk.sh testL2     
Error: please provide the resource name and the type of CDK constructor

Usage:
./cdk.sh   "<RESOURCE NAME>" "<L1|L2|L3>"

Example:
./cdk.sh cluster L1
```

</details>

### CODE REVIEW

File: cdk/cdk.sh [hunk 1]
```diff
@@ -1,56 +1,103 @@
...
-if [[ -n "$resources" ]];
-then
-  source_dir=${resources};
-else
-  echo "Generating for all resources."
-  source_dir='*';
-  dir="../cfn-resources/$source_dir"
+# How to use it:
+#   1. CDK L1:  ./cdk.sh "<RESOURCE NAME>" L1
+#   1. CDK L1:  ./cdk.sh "<RESOURCE NAME>" L1
```
Comment (tibulca) [inline cdk/cdk.sh:23]:
```suggestion
#   1. CDK L1:  ./cdk.sh "<RESOURCE NAME>" L1
```

File: cdk/cdk.sh [hunk 2]
```diff
@@ -1,56 +1,103 @@
...
+#   2. CDK L2:  ./cdk/sh "<RESOURCE NAME>" L2
+#   4. CDK L3:  ./cdk/sh "<RESOURCE NAME>" L3
+
+set -euo pipefail
+
+
+_print_usage() {
+	echo
+  echo 'Usage:'
+	echo './cdk.sh   "<RESOURCE NAME>" "<L1|L2|L3>"'
```
Comment (tibulca) [inline cdk/cdk.sh:33]:
```suggestion
  echo
  echo 'Usage:'
  echo './cdk.sh   "<RESOURCE NAME>" "<L1|L2|L3>"'
```

File: cdk/cdk.sh [hunk 3]
```diff
@@ -1,56 +1,103 @@
...
+if [ "$cdk_type" = "L3" ]; then
+  echo "Generating L3 CDK resource"
+  mkdir "l3-cdk-resources/${resource}"
+  pushd "l3-cdk-resources/${resource}"
+
+  npx projen new awscdk-construct --npm-access "public" --author "MongoDBAtlas" --author-name "MongoDBAtlas" --docgen true --sample-code false --name "@mongodbatlas-awscdk-l3/${resource}" --author-address 'https://mongodb.com' --cdk-version '2.1.0' --default-release-branch 'master' --major-version 1 --release-to-npm true --repository-url 'https://github.com/mongodb/mongodbatlas-cloudformation-resources.git' --description "Retrieves or creates ${resource} in any given Atlas organization" --keywords {'cdk','awscdk','aws-cdk','cloudformation','cfn','extensions','constructs','cfn-resources','cloudformation-registry','l3','mongodb','atlas',"$resource"}
+  rm -rf .git
+
+  popd
+fi
```
Comment (tibulca) [inline cdk/cdk.sh:101]: this code seems to be almost identical (the only diff is l1/l2/l3) with 2 more sections above.
Can we extract common code into a function?
Example: https://github.com/10gen/mms/blob/43a6f3534ad8c23591b2a063e25462296c2abac7/scripts/evergreen/openapi/save_openapi_files.sh#L18

File: cdk/cdk.sh (hunk 2 above)
Comment (andreaangiolillo) [reply cdk/cdk.sh]: updated

**Review by maastha — APPROVED** [review]

File: cdk/cdk.sh (hunk 3 above)
Comment (andreaangiolillo) [reply cdk/cdk.sh]: Updated

**Pull request merged**